crates/miden-objects/src/transaction/inputs.rs

```diff
@@ -536,144 +535,93 @@ pub fn validate_account_seed(
     }
 }
 
-// FOREIGN ACCOUNT INPUTS
+// ACCOUNT INPUTS
 // ================================================================================================
 
-/// Contains information about a foreign account, with everything required to execute its code.
+/// Contains information about an account, with everything required to execute a transaction.
+///
+/// `AccountInputs` combines a partial account representation with the merkle proof that verifies
+/// the account's inclusion in the account tree. The partial account should contain verifiable
+/// access to the segements of the state of the account of which the transaction will make use
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub struct ForeignAccountInputs {
-    /// Account header of the foreign account.
-    account_header: AccountHeader,
-    /// Header information about the account's storage.
-    storage_header: AccountStorageHeader,
-    /// Code associated with the account.
-    account_code: AccountCode,
+pub struct AccountInputs {
+    /// Partial representation of the account's state.
+    partial_account: PartialAccount,
     /// Proof of the account's inclusion in the account tree for this account's state commitment.
     witness: AccountWitness,
-    /// Storage SMT proof for storage map values that the transaction will access.
-    storage_map_proofs: Vec<SmtProof>,
 }
 
-impl ForeignAccountInputs {
-    /// Creates a new [`ForeignAccountInputs`]
-    pub fn new(
-        account_header: AccountHeader,
-        storage_header: AccountStorageHeader,
-        account_code: AccountCode,
-        witness: AccountWitness,
-        storage_map_proofs: Vec<SmtProof>,
-    ) -> ForeignAccountInputs {
-        ForeignAccountInputs {
-            account_header,
-            storage_header,
-            account_code,
-            witness,
-            storage_map_proofs,
-        }
+impl AccountInputs {
+    /// Creates a new instance of `AccountInputs` with the specified partial account and witness.
+    pub fn new(partial_account: PartialAccount, witness: AccountWitness) -> AccountInputs {
+        AccountInputs { partial_account, witness }
     }
 
-    /// Returns the account's [`AccountId`]
+    /// Returns the account ID.
     pub fn id(&self) -> AccountId {
-        self.account_header.id()
+        self.partial_account.id()
     }
 
-    /// Returns the account's [`AccountHeader`]
-    pub fn account_header(&self) -> &AccountHeader {
-        &self.account_header
+    /// Returns a reference to the partial account representation.
+    pub fn partial_account(&self) -> &PartialAccount {
+        &self.partial_account
     }
```

Comment:
nit: this could also probably be just `account()` - though, given that we expose pretty much all parts of the account separately - do we need this at all?